pkg/model/components/etcd.go

```diff
@@ -32,13 +34,31 @@ var _ loader.OptionsBuilder = &EtcdOptionsBuilder{}
 func (b *EtcdOptionsBuilder) BuildOptions(o interface{}) error {
 	spec := o.(*kops.ClusterSpec)
 
+	imageVersion := ""
 	// @check the version are set and if not preset the defaults
 	for _, x := range spec.EtcdClusters {
 		// @TODO if nothing is set, set the defaults. At a late date once we have a way of detecting a 'new' cluster
 		// we can default all clusters to v3
 		if x.Version == "" {
 			x.Version = "2.2.1"
 		}
+
+		if imageVersion != "" {
+			if imageVersion != x.Version {
+				return fmt.Errorf("you cannot have two different versions of etcd %q and %q set", x.Version, imageVersion)
+			}
+		} else {
+			imageVersion = x.Version
+		}
+	}
+
+	// TODO we do not have a API place to put this.  Do we use version or get another
+	// TODO API value?
+	// TODO code is duplicated in nodeup
+	image := fmt.Sprintf("gcr.io/google_containers/etcd:%s", imageVersion)
```

Comment:
This is validating that the assetbuilder is working?  Maybe we should explicitly put the etcd image into the nodeup spec.

Reply:
no this is building the URL for image builder ... 

Yes, we should put it in the nodeup spec.  This PR or another?  I would say in another PR.  But shouldn't it be one container in the etcd spec??  I am seriously confused why we have the version where we have it :(

Reply:
Do we want the etcd container in the clusterspec and nodeup api?  Or can we do that in another PR?

Reply:
We have the asset builder here do you want to use that or an API value?